language/libra-vm/src/libra_vm.rs

```diff
@@ -459,15 +459,15 @@ pub fn txn_effects_to_writeset_and_events_cached<C: AccessPathCache>(
 
     let ws = WriteSetMut::new(ops)
         .freeze()
-        .map_err(|_| VMStatus::Error(StatusCode::DATA_FORMAT_ERROR))?;
+        .map_err(|_| VMStatus::Error(StatusCode::UNKNOWN_INVARIANT_VIOLATION_ERROR))?;
 
     let events = effects
         .events
         .into_iter()
         .map(|(guid, seq_num, ty_tag, ty_layout, val)| {
             let msg = val
                 .simple_serialize(&ty_layout)
-                .ok_or_else(|| VMStatus::Error(StatusCode::DATA_FORMAT_ERROR))?;
+                .ok_or_else(|| VMStatus::Error(StatusCode::UNKNOWN_INVARIANT_VIOLATION_ERROR))?;
             let key = EventKey::try_from(guid.as_slice())
                 .map_err(|_| VMStatus::Error(StatusCode::EVENT_KEY_MISMATCH))?;
```

Comment:
@dariorussi @tnowacki @tzakian It seems to me that it's technically possible for someone to emit an event from Move with a bad guid and trigger this and therefore I'm keeping it as a user error. Let me know if this sounds correct.

On the other hand, I think events in their current form are definitely a Libra concept and will need to be redesigned when we separate Move from Libra. I feel like instead of having
```
native fun write_to_event_store<T: copyable>(guid: vector<u8>, count: u64, msg: T);
```
we'll have
```
native fun write_to_event_store<T: Copyable>(msg: T);
```
In the case of Libra, we will just have another wrapper function that creates an emits a `struct LibraEvent<T: copyable> { guid: ..., seq_num: ..., msg: T }`.

However it's also possible this will simply no longer be a problem if we happen to find a way to solve the general issues with native functions that are closely associated with the VM runtime.

Reply:
more of a question for @sblackshear, @runtian-zhou and the others mentioned that for me :)

Reply:
> and therefore I'm keeping it as a user error. Let me know if this sounds correct.

`EVENT_KEY_MISMATCH` is an invariant violation. So if a user can hit this, this would be very bad. 
Can you write a test where this is hit?

Reply:
Oh yeah, NVM. `fun write_to_event_store` is private and other Move code needs to use `fun emit_event`, which will always generate a valid guid if working properly. So in this case the infeasibility to trigger this error is guaranteed by the Move implementation of the Libra framework.